docs/designs/operator.md

```diff
@@ -0,0 +1,201 @@
+  - Ensure the operator's desired state is always in sync with the actual cluster state, supporting both declarative and imperative workflows.
+
+- **Incremental, Granular Reconciliation:**  
+  - The operator proceeds in highly granular steps, maintaining subresource conditions for each phase (e.g., ADX cluster ready, database ready, ingestor ready, collector ready).
+  - Each phase is only attempted after all dependencies are successfully reconciled.
+  - Status and conditions are updated at each step to reflect progress and issues.
+
+- **Resource Cleanup on Deletion:**  
+  - When an Operator CRD is deleted, the operator will also delete all managed resources, including ASO CRDs (for ADX clusters and databases) and adx-mon component manifests (ingestor, collector, alerter), ensuring a clean teardown of the stack.
+
+---
+
+## Multi-Cluster and Federation Support
+
+- The operator supports deployment in different Kubernetes clusters for federated scenarios:
+  - **Central Ingestor Cluster:**  
+    - Deploys ingestor components (and optionally ADX resources).
+    - Receives events from multiple remote collector clusters.
+  - **Collector Clusters:**  
+    - Deploys only collector components.
+    - Forwards events to a central ingestor cluster.
+    - Requires configuration of ingestor endpoints and authentication.
+
+- The CRD spec determines the deployment mode:
+  - If only `collector` is specified, the operator assumes this is a collector-only cluster and requires `ingestor` connection details.
+  - If `ingestor` and `adx` are specified, the operator will bootstrap the full stack and expose endpoints for remote collectors.
+
+---
+
+## CRD Design
+
+The `Operator` CRD is the entry point for managing an adx-mon cluster. It should support:
+
+- **Minimal configuration for quick bootstrap:**  
+  - Defaults for all images:  
+    - `ghcr.io/azure/adx-mon/collector:latest`
+    - `ghcr.io/azure/adx-mon/ingestor:latest`
+    - `ghcr.io/azure/adx-mon/alerter:latest`
+  - Default to a single ADX cluster and database.
+  - Default to 1 replica for each component.
+
+- **Customizable fields:**  
+  - Images for each component.
+  - Number of ingestor replicas.
+  - ADX cluster/database names and configuration.
+  - Support for multiple ADX clusters, each with multiple databases.
+  - Each database specifies a telemetry type: `Logs`, `Metrics`, or `Traces`.
+  - Cluster connection info, including endpoint and MSI client-id if using managed identity.
+  - Optional: advanced overrides for component manifests.
+  - **Federation fields:**  
+    - For collector-only clusters: specify `ingestor` connection details (endpoint, authentication).
+    - For ingestor clusters: expose endpoints for remote collectors.
+
+- **Status fields:**  
+  - Reflect the current state and subresource conditions (e.g., ADX cluster ready, database ready, ingestor ready, collector ready).
+
+### Example CRD
```

Comment:
I like the feel of this.  Very simple and straight-forward.    We should explore what more complicated setup options we might need to support and see if it would make this CRD too complicated.  For example, defining singleton collectors or more complicated collector configmaps.